public/store/reducers/opensearch_reducer.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { getRouteService } from '../../services';
+import { IIndex } from '../../../common';
+
+const initialState = {
+  loading: false,
+  errorMessage: '',
+  indices: {} as { [key: string]: IIndex },
+};
+
+const OPENSEARCH_PREFIX = 'opensearch';
+const FETCH_INDICES_ACTION = `${OPENSEARCH_PREFIX}/fetchIndices`;
+
+export const fetchIndices = createAsyncThunk(
+  FETCH_INDICES_ACTION,
+  async (pattern?: string) => {
+    // defaulting to fetch everything except system indices (starting with '.')
+    const patternString = pattern || '*,-.*';
+    const response = getRouteService().fetchIndices(patternString);
+    return response;
+  }
+);
+
+const opensearchSlice = createSlice({
+  name: OPENSEARCH_PREFIX,
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchIndices.pending, (state, action) => {
+        state.loading = true;
+      })
+      .addCase(fetchIndices.fulfilled, (state, action) => {
+        const indicesMap = new Map<string, IIndex>();
```

Comment:
Should we declare the map on top of createSlice on line 28?

Reply:
That was my first attempt :) but it isn't serializable so I converted to a plain JS object

Reply:
ref: https://stackoverflow.com/questions/67763839/why-is-a-map-in-js-considered-non-serializable-in-redux